scipy/interpolate/_ndgriddata.py

```diff
@@ -30,9 +30,9 @@ class NearestNDInterpolator(NDInterpolatorBase):
 
     Parameters
     ----------
-    x : (npoints, ndims) 2-D ndarray of floats
+    x : (npoints, ndims) n-D ndarray of floats
         Data point coordinates.
-    y : (npoints, ) 1-D ndarray of float or complex
+    y : (npoints, ...) n-D ndarray of floats or complex
```

Comment:
OK, so the original was papering over a detail that `y` can have trailing dimensions (BTW, I'm not sure LinearNDInterpolator and CT can). 
The new version is plain confusing, my mind starts racing to figure out what is `n` in  `n-D`. Essentially, what is going on here is that `y` is a 1D array of data values or a stack of 1D arrays of data values, correct? I'd probably say something along these lines then (ML people like to talk about batch dimensions).

Reply:
I realized my response to this didn't go through. Yeah, you're right - I've fixed this in latest commit. Looks like Linear and CT can actually. There's some voodoo going on in NDInterpolatorBase's _set_values, where it'll recast values to be 2D (npoints, nvalues) for all interoplators that subclass.